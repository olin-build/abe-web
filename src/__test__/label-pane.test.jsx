import React from 'react';
import renderer from 'react-test-renderer';
import LabelPane from '../components/label-pane';

describe('LabelPane', () => {
  const labels = {
    first: {
      name: 'first',
      id: 'label-1',
      color: 'red',
      description: 'first label',
    },
    second: {
      name: 'second',
      id: 'label-2',
      color: 'green',
      description: 'second label',
      protected: true,
    },
  };
  test('matches baseline', () => {
    const jsx = <LabelPane possibleLabels={labels} />;
    const component = renderer.create(jsx);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('shows selected labels', () => {
    const jsx = <LabelPane possibleLabels={labels} selectedLabels={['first']} />;
    const component = renderer.create(jsx);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('hides unselected labels', () => {
    const jsx = (
      <LabelPane possibleLabels={labels} selectedLabels={['first']} showUnselected={false} />
    );
    const component = renderer.create(jsx);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('disables protected labels', () => {
    const jsx = <LabelPane disableProtectedLabels possibleLabels={labels} />;
    const component = renderer.create(jsx);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('not editable', () => {
    const component = renderer.create(<LabelPane editable={false} possibleLabels={labels} selectedLabels={['first']} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('sorts labels', () => {
    const labels2 = {
      0: { name: 'featured' },
      1: { name: 'a-label-2' },
      2: { name: 'a-label-1' },
      3: { name: 'a-label-4', default: true },
      4: { name: 'a-label-3', default: true },
    };
    // These should come out in order 'featured', 'a-label-3', 'a-label-4', 'a-label-1', a-label-2'.
    // Initially verified from manual inspection of the snapshot.
    Object.values(labels2).forEach((label, ix) => {
      label.id = `id-${ix}`; // eslint-disable-line no-param-reassign
    });
    const component = renderer.create(<LabelPane possibleLabels={labels2} showUnselected />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
