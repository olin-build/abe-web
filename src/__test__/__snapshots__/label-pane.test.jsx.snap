// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelPane matches snapshot 1`] = `
<div
  className="css-class"
>
  <style
    type="text/css"
  >
    .label.label-label-1.selected{background-color:red}
    .label.label-label-1.selected:not(.no-hover):hover{background-color:white;border-color:red;color:red}
    .label.label-label-1:not(.selected){background-color:white;border-color:red;color:red}
    .label.label-label-1:not(.no-hover):hover{background-color:red;color:white}
    .label.label-label-2.selected{background-color:green}
    .label.label-label-2.selected:not(.no-hover):hover{background-color:white;border-color:green;color:green}
    .label.label-label-2:not(.selected){background-color:white;border-color:green;color:green}
    .label.label-label-2:not(.no-hover):hover{background-color:green;color:white}
  </style>
  <div
    className="label-selector-list"
  >
    <button
      className="button  label label-label-1 "
      id="label-1"
      onClick={[Function]}
      title="first label"
      type="button"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      first
    </button>
    <button
      className="button  label label-label-2 "
      id="label-2"
      onClick={[Function]}
      title="second label"
      type="button"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      second
    </button>
  </div>
</div>
`;

exports[`LabelPane not editable 1`] = `
<div
  className="css-class"
>
  <style
    type="text/css"
  >
    .label.label-label-1.selected{background-color:red}
    .label.label-label-1.selected:not(.no-hover):hover{background-color:white;border-color:red;color:red}
    .label.label-label-1:not(.selected){background-color:white;border-color:red;color:red}
    .label.label-label-1:not(.no-hover):hover{background-color:red;color:white}
    .label.label-label-2.selected{background-color:green}
    .label.label-label-2.selected:not(.no-hover):hover{background-color:white;border-color:green;color:green}
    .label.label-label-2:not(.selected){background-color:white;border-color:green;color:green}
    .label.label-label-2:not(.no-hover):hover{background-color:green;color:white}
  </style>
  <div
    className="label-selector-list"
  >
    <span
      className="label label-label-1 selected"
      id="label-1"
      title="first label"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      first
    </span>
    <span
      className="label label-label-2 "
      id="label-2"
      title="second label"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      second
    </span>
  </div>
</div>
`;

exports[`LabelPane sorts labels 1`] = `
<div
  className=""
>
  <style
    type="text/css"
  >
    .label.label-id-0.selected{background-color:undefined}
    .label.label-id-0.selected:not(.no-hover):hover{background-color:white;border-color:undefined;color:undefined}
    .label.label-id-0:not(.selected){background-color:white;border-color:undefined;color:undefined}
    .label.label-id-0:not(.no-hover):hover{background-color:black;color:white}
    .label.label-id-4.selected{background-color:undefined}
    .label.label-id-4.selected:not(.no-hover):hover{background-color:white;border-color:undefined;color:undefined}
    .label.label-id-4:not(.selected){background-color:white;border-color:undefined;color:undefined}
    .label.label-id-4:not(.no-hover):hover{background-color:black;color:white}
    .label.label-id-3.selected{background-color:undefined}
    .label.label-id-3.selected:not(.no-hover):hover{background-color:white;border-color:undefined;color:undefined}
    .label.label-id-3:not(.selected){background-color:white;border-color:undefined;color:undefined}
    .label.label-id-3:not(.no-hover):hover{background-color:black;color:white}
    .label.label-id-2.selected{background-color:undefined}
    .label.label-id-2.selected:not(.no-hover):hover{background-color:white;border-color:undefined;color:undefined}
    .label.label-id-2:not(.selected){background-color:white;border-color:undefined;color:undefined}
    .label.label-id-2:not(.no-hover):hover{background-color:black;color:white}
    .label.label-id-1.selected{background-color:undefined}
    .label.label-id-1.selected:not(.no-hover):hover{background-color:white;border-color:undefined;color:undefined}
    .label.label-id-1:not(.selected){background-color:white;border-color:undefined;color:undefined}
    .label.label-id-1:not(.no-hover):hover{background-color:black;color:white}
  </style>
  <div
    className="label-selector-list"
  >
    <button
      className="button  label label-id-0 "
      id="id-0"
      onClick={[Function]}
      title={undefined}
      type="button"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      featured
    </button>
    <button
      className="button  label label-id-4 "
      id="id-4"
      onClick={[Function]}
      title={undefined}
      type="button"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      a-label-3
    </button>
    <button
      className="button  label label-id-3 "
      id="id-3"
      onClick={[Function]}
      title={undefined}
      type="button"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      a-label-4
    </button>
    <button
      className="button  label label-id-2 "
      id="id-2"
      onClick={[Function]}
      title={undefined}
      type="button"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      a-label-1
    </button>
    <button
      className="button  label label-id-1 "
      id="id-1"
      onClick={[Function]}
      title={undefined}
      type="button"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      a-label-2
    </button>
  </div>
</div>
`;

exports[`LabelPane with selected labels 1`] = `
<div
  className="css-class"
>
  <style
    type="text/css"
  >
    .label.label-label-1.selected{background-color:red}
    .label.label-label-1.selected:not(.no-hover):hover{background-color:white;border-color:red;color:red}
    .label.label-label-1:not(.selected){background-color:white;border-color:red;color:red}
    .label.label-label-1:not(.no-hover):hover{background-color:red;color:white}
    .label.label-label-2.selected{background-color:green}
    .label.label-label-2.selected:not(.no-hover):hover{background-color:white;border-color:green;color:green}
    .label.label-label-2:not(.selected){background-color:white;border-color:green;color:green}
    .label.label-label-2:not(.no-hover):hover{background-color:green;color:white}
  </style>
  <div
    className="label-selector-list"
  >
    <button
      className="button  label label-label-1 selected"
      id="label-1"
      onClick={[Function]}
      title="first label"
      type="button"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      first
    </button>
    <button
      className="button  label label-label-2 "
      id="label-2"
      onClick={[Function]}
      title="second label"
      type="button"
    >
      <span
        className="ion-pricetag"
      >
         
      </span>
      second
    </button>
  </div>
</div>
`;
